File: Pony_Racing/core/src/com/esotericsoftware/spine/attachments/AttachmentLoader.java


package com.esotericsoftware.spine.attachments;

import com.esotericsoftware.spine.Skin;

public interface AttachmentLoader {
    /**
     * @return May be null to not load any attachment.
     */
    RegionAttachment newRegionAttachment(Skin skin, String name, String path);

    /**
     * @return May be null to not load any attachment.
     */
    MeshAttachment newMeshAttachment(Skin skin, String name, String path);

    /**
     * @return May be null to not load any attachment.
     */
    BoundingBoxAttachment newBoundingBoxAttachment(Skin skin, String name);
}

File: Pony_Racing/core/src/com/esotericsoftware/spine/attachments/AttachmentLoader.kt
package com.esotericsoftware.spine.attachments

import com.esotericsoftware.spine.Skin

interface AttachmentLoader {
    /**
     * @return May be null to not load any attachment.
     */
    fun newRegionAttachment(skin: Skin, name: String, path: String): RegionAttachment?

    /**
     * @return May be null to not load any attachment.
     */
    fun newMeshAttachment(skin: Skin, name: String, path: String): MeshAttachment?

    /**
     * @return May be null to not load any attachment.
     */
    fun newBoundingBoxAttachment(skin: Skin, name: String): BoundingBoxAttachment?
}